apps/os/backend/agent/do-tools.ts

```diff
@@ -88,3 +89,24 @@ export function doToolToRuntimeJsonSchema(_value: unknown) {
   };
   return runtimeJsonSchema;
 }
+
+export function createDOToolFactory<T extends ReturnType<typeof defineDOTools>>(definitions: T) {
```

Comment:
I'd use the same generic as defineDOTools `T extends Record<string, DOToolDef<any, any>>`